lib/dl_core/dl_core/us_manager/factory.py

```diff
@@ -54,6 +55,13 @@ def get_master_auth_context(self) -> USAuthContextMaster:
         assert self.us_master_token is not None, "US master token must be set in factory to create USAuthContextMaster"
         return USAuthContextMaster(us_master_token=self.us_master_token)
 
+    def get_master_auth_context_from_headers(self) -> USAuthContextMaster:
+        us_master_token = flask.request.headers.get(DLHeadersCommon.US_MASTER_TOKEN.value)
+        assert (
+            us_master_token is not None
+        ), f"US master token must be set in header {DLHeadersCommon.US_MASTER_TOKEN.value} to create USAuthContextMaster"
```

Comment:
If the header is missing, this assert will result in a 500, won't it? I'd suggest raising some kind of exception instead of assert here